website/orders/static/orders/js/asynch-update-orders.js

```diff
@@ -1,126 +0,0 @@
-
-String.prototype.capitalize = function() {
-    return this.charAt(0).toUpperCase() + this.slice(1);
-};
-
-function get_orders(data_url, csrf_token, callback_ok, callback_error, /*, args */) {
-    let args = Array.prototype.slice.call(arguments, 4);
-    jQuery(function($) {
-        let data = {
-            'csrfmiddlewaretoken': csrf_token,
-        };
-        $.ajax({type: 'POST', url: data_url, data, dataType:'json', asynch: true, success:
-            function(data) {
-                args.unshift(data.data);
-                callback_ok.apply(this, args);
-            }}).fail(function() {
-                args.unshift("Error while getting order data.");
-                callback_error.apply(this, args);
-            });
-        }
-    )
-}
-
-function create_checkbox(value, order_id, property, class_list) {
-    let element = document.createElement('input');
-    element.type = 'button';
-    element.value = property.capitalize();
-    element.setAttribute('onclick',"update_order(this, '" + order_id + "', '" + property + "', update_checkbox, update_checkbox_error)");
-    element.classList.add('btn');
-    if (value) {
-        element.classList.add('btn-success');
-    }
-    else {
-        element.classList.add('btn-danger');
-    }
-    for (let i = 0; i < class_list.length; i++) {
-        element.classList.add(class_list[i]);
-    }
-    return element;
-}
-
-function update_order_error(error, element) {
-
-}
-
-function update_order_data(orders, element) {
-    let base = document.createElement('ol');
-    base.classList.add('item-list');
-    for (let i = 0; i < orders.length; i++) {
-        let list_item_wrapper = create_element('li', ['pt-2', 'pb-2'], '');
```

Comment:
In my opinion it would be way easier to understand to not use `create_elements` but let Django create the orders in a html template with a template tag, and only use javascript to refresh template tag.

Reply:
We could implement that later as this works pretty nicely now.

Reply:
In that case, let's put this in a new issue for the backlog